### PR TITLE
Better description for Shrink Database button

### PR DESCRIPTION
**Pull request by Saibamen**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [ ] I have read and understand the pull request rules.

# Description

As discussed in https://github.com/louislam/uptime-kuma/issues/2470#issuecomment-1953725902

Previous description suggested that you need to recreate your database from scratch.

## Type of change

Please delete any options that are not relevant.

- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

image

### CODE REVIEW

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Changing the meaning of a translation is not great, as this leaves the translators unaware of this change.

A better approach would be to introduce a translation string and remove the current one.
This also allows using the [`i18n-t`-component](https://vue-i18n.intlify.dev/guide/advanced/component) to display `VACUUM`/`AUTO_VACUUM` as `<code>VACUUM</code>`/`<code>AUTO_VACUUM</code>`
Here is the relevant line:
https://github.com/louislam/uptime-kuma/blob/cf2d603e277d18e96906ac36743ec4e7d58b8aad/src/components/settings/MonitorHistory.vue#L35

**Comment by Saibamen** [comment]
> Changing the meaning of a translation is not great, as this leaves the translators unaware of this change.

I translated via Weblate and I bet that I translated few already translated i18n keys with changed English source.

But I like to have HTML's `<code>`. I will change that today. Thanks

**Comment by Saibamen** [comment]
Done

image

File: src/components/settings/MonitorHistory.vue [hunk 1]
```diff
@@ -32,7 +32,14 @@
                 <button class="btn btn-outline-info me-2" @click="shrinkDatabase">
                     {{ $t("Shrink Database") }} ({{ databaseSizeDisplay }})
                 </button>
-                <div class="form-text mt-2 mb-4 ms-2">{{ $t("shrinkDatabaseDescription") }}</div>
+                <i18n-t tag="div" keypath="shrinkDatabaseDescriptionNew" class="form-text mt-2 mb-4 ms-2">
+                    <template v-slot:command1>
```
Comment (CommanderStorm) [inline src/components/settings/MonitorHistory.vue:36]:
```suggestion
                    <template #vacuum>
```

File: src/components/settings/MonitorHistory.vue [hunk 2]
```diff
@@ -32,7 +32,14 @@
...
+                        <code>VACUUM</code>
+                    </template>
+                    <template v-slot:command2>
```
Comment (CommanderStorm) [inline src/components/settings/MonitorHistory.vue:39]:
```suggestion
                    <template #auto_vacuum>
```

File: src/lang/en.json [hunk 3]
```diff
@@ -818,5 +817,6 @@
     "showCertificateExpiry": "Show Certificate Expiry",
     "noOrBadCertificate": "No/Bad Certificate",
     "gamedigGuessPort": "Gamedig: Guess Port",
-    "gamedigGuessPortDescription": "The port used by Valve Server Query Protocol may be different from the client port. Try this if the monitor cannot connect to your server."
+    "gamedigGuessPortDescription": "The port used by Valve Server Query Protocol may be different from the client port. Try this if the monitor cannot connect to your server.",
+    "shrinkDatabaseDescriptionNew": "Trigger database {command1} for SQLite. {command2} is already enabled but this does not defragment the database nor repack individual database pages the way that the {command1} command does."
```
Comment (CommanderStorm) [inline src/lang/en.json:821]:
```suggestion
    "shrinkDatabaseDescription": "Trigger database {vacuum} for SQLite. {auto_vacuum} is already enabled, but this does not defragment the database nor repack individual database pages the way that the {vacuum} command does."
```

File: src/components/settings/MonitorHistory.vue [hunk 4]
```diff
@@ -32,7 +32,14 @@
                 <button class="btn btn-outline-info me-2" @click="shrinkDatabase">
                     {{ $t("Shrink Database") }} ({{ databaseSizeDisplay }})
                 </button>
-                <div class="form-text mt-2 mb-4 ms-2">{{ $t("shrinkDatabaseDescription") }}</div>
+                <i18n-t tag="div" keypath="shrinkDatabaseDescriptionNew" class="form-text mt-2 mb-4 ms-2">
```
Comment (CommanderStorm) [inline src/components/settings/MonitorHistory.vue:35]:
```suggestion
                <i18n-t tag="div" keypath="shrinkDatabaseDescription" class="form-text mt-2 mb-4 ms-2">
```

**Review by CommanderStorm — COMMENTED** [review]
I have made a few changes.
Given that you touched the other translation files, this PR can only be merged directly after merging https://github.com/louislam/uptime-kuma/pull/4436 and https://github.com/louislam/uptime-kuma/pull/4394 to avoid merge conflicts being created in Weblate.

**Comment by CommanderStorm** [comment]
There may be other merge conflicts that come up. v2.0 does support mariadb next to sqlite, but said db still needs to be integrated here as well

File: src/components/settings/MonitorHistory.vue (hunk 4 above)
Comment (Saibamen) [reply src/components/settings/MonitorHistory.vue]: `shrinkDatabaseDescription` is the old i18n key. I thought you wanted a completely new key:

> A better approach would be to introduce a translation string and remove the current one.

**Comment by Saibamen** [comment]
>There may be other merge conflicts that come up. v2.0 does support mariadb
This PR is for 1.23.x only

File: src/components/settings/MonitorHistory.vue (hunk 4 above)
Comment (CommanderStorm) [reply src/components/settings/MonitorHistory.vue]: given that the other keys are all removed in this version, there is no need to switch to a different key.

Comment (Saibamen) [reply src/components/settings/MonitorHistory.vue]: ok, I moved key to previous place

Comment (CommanderStorm) [reply src/components/settings/MonitorHistory.vue]: Thanks, that is something that I can't do just by code-reviews ^^

**Comment by louislam** [comment]
Since merging conflicts on Weblate is painful, as the git repo is read only. And removing the key is likely causing this problem. 

I think we need to find another way to "deprecated" this key instead of removing it.

**Comment by CommanderStorm** [comment]
I think this should not cause a merge-conflict if merged directly after https://github.com/louislam/uptime-kuma/pull/4436

What might indeed be problematic is that users can't supply translations for `v1.23.X`, only for `v2.X`
=> this change will never get translated.

**Comment by louislam** [comment]
> I think this should not cause a merge-conflict if merged directly after #4436 Am I missing some context?
> 

Since it is for the 1.23.X branch, it will happen when I merge the 1.23.x branch to master, I believe. 

Of course I can merge the 1.23.X branch to master after #4436, but it is kind of "soft-locked".

> What might indeed be problematic is that users can't supply translations for `v1.23.X`, only for `v2.X` => this change will never get translated.

You are right. And 1.23.X is indeed not translatable in the current stage, as Weblate don't support for multiple branches. (Or I don't know how to set up)

For simplicity, I would suggest this pr should be merged into the master branch directly and also be merged after #4436.

**Comment by Saibamen** [comment]
> Weblate don't support for multiple branches

@louislam:
https://docs.weblate.org/en/latest/faq.html#how-do-i-translate-several-branches-at-once

**Comment by Saibamen** [comment]
ping

**Comment by CommanderStorm** [comment]
> For simplicity, I would suggest this pr should be merged into the master branch directly and also be merged after #4436.

I have rebase the PR in https://github.com/louislam/uptime-kuma/pull/4814 (different PR due to githubs permission system ^^)

**Pull request closed without merging**